[2025-06-05T05:00:01.256+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: comments_ingestion.fetch_and_store_comments_task scheduled__2025-06-05T04:30:00+00:00 [queued]>
[2025-06-05T05:00:01.271+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: comments_ingestion.fetch_and_store_comments_task scheduled__2025-06-05T04:30:00+00:00 [queued]>
[2025-06-05T05:00:01.272+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-06-05T05:00:01.299+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): fetch_and_store_comments_task> on 2025-06-05 04:30:00+00:00
[2025-06-05T05:00:01.306+0000] {standard_task_runner.py:60} INFO - Started process 603 to run task
[2025-06-05T05:00:01.313+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'comments_ingestion', 'fetch_and_store_comments_task', 'scheduled__2025-06-05T04:30:00+00:00', '--job-id', '39', '--raw', '--subdir', 'DAGS_FOLDER/api_to_db_dag_commments.py', '--cfg-path', '/tmp/tmpbsay8k1i']
[2025-06-05T05:00:01.317+0000] {standard_task_runner.py:88} INFO - Job 39: Subtask fetch_and_store_comments_task
[2025-06-05T05:00:01.386+0000] {task_command.py:423} INFO - Running <TaskInstance: comments_ingestion.fetch_and_store_comments_task scheduled__2025-06-05T04:30:00+00:00 [running]> on host e4f5e6f9211f
[2025-06-05T05:00:01.495+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='comments_ingestion' AIRFLOW_CTX_TASK_ID='fetch_and_store_comments_task' AIRFLOW_CTX_EXECUTION_DATE='2025-06-05T04:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-06-05T04:30:00+00:00'
[2025-06-05T05:00:01.497+0000] {api_to_db_dag_commments.py:37} INFO - Đang gọi API: https://app.circle.so/api/admin/v2/comments?page=1&per_page=10
[2025-06-05T05:00:06.369+0000] {api_to_db_dag_commments.py:40} INFO - Đã lấy 45 bản ghi từ API (nhiều trang).
[2025-06-05T05:00:06.370+0000] {api_to_db_dag_commments.py:47} INFO - Đang kết nối đến cơ sở dữ liệu với Conn ID: QUANG
[2025-06-05T05:00:06.381+0000] {base.py:83} INFO - Using connection ID 'QUANG' for task execution.
[2025-06-05T05:00:06.535+0000] {api_to_db_dag_commments.py:53} INFO - Kết nối cơ sở dữ liệu thành công.
[2025-06-05T05:00:06.574+0000] {api_to_db_dag_commments.py:114} ERROR - Lỗi khi thao tác với database: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`sql12783027`.`comments`, CONSTRAINT `comments_ibfk_2` FOREIGN KEY (`post_id`) REFERENCES `posts` (`post_id`))')
[2025-06-05T05:00:06.577+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/api_to_db_dag_commments.py", line 103, in fetch_and_store_comments
    cur.executemany(insert_sql, values_to_insert)
  File "/home/airflow/.local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 241, in executemany
    return self._do_execute_many(
  File "/home/airflow/.local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 275, in _do_execute_many
    rows += self.execute(sql + postfix)
  File "/home/airflow/.local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 179, in execute
    res = self._query(mogrified_query)
  File "/home/airflow/.local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 330, in _query
    db.query(q)
  File "/home/airflow/.local/lib/python3.8/site-packages/MySQLdb/connections.py", line 257, in query
    _mysql.connection.query(self, query)
MySQLdb.IntegrityError: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`sql12783027`.`comments`, CONSTRAINT `comments_ibfk_2` FOREIGN KEY (`post_id`) REFERENCES `posts` (`post_id`))')
[2025-06-05T05:00:06.599+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=comments_ingestion, task_id=fetch_and_store_comments_task, execution_date=20250605T043000, start_date=20250605T050001, end_date=20250605T050006
[2025-06-05T05:00:06.637+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 39 for task fetch_and_store_comments_task ((1452, 'Cannot add or update a child row: a foreign key constraint fails (`sql12783027`.`comments`, CONSTRAINT `comments_ibfk_2` FOREIGN KEY (`post_id`) REFERENCES `posts` (`post_id`))'); 603)
[2025-06-05T05:00:06.680+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-05T05:00:06.713+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
